pylint_nautobot/constants.py

```diff
@@ -0,0 +1,145 @@
+            "nb-use-fields-all",
+            "Defining `fields = '__all__'` in a model serializer's Meta class is a Django convention that automatically "
+            "includes all fields from the associated model. This approach is more maintainable because it avoids having "
+            "to explicitly list each field, reducing the risk of errors and inconsistencies when the model is updated.",
+        ),
+    }
+    C4272: dict[str, MessageDefinitionTuple] = {
+        "C4272": (
+            "Use `q = SearchFilter` instead of django_filters.CharFilter and the custom `search` function.",
+            "nb-no-char-filter-q",
+            "Nautobot provides a `SearchFilter` class that uses MappedPredicates to provide a more flexible search experience. "
+            "This should be used in place of `django_filters.CharFilter` and no longer requires the custom `search` function.",
+        ),
+    }
+    C4273: dict[str, MessageDefinitionTuple] = {
+        "C4273": (
+            "Use `q = SearchFilter` instead.",
+            "nb-use-search-filter",
+            "Nautobot provides a `SearchFilter` class that uses MappedPredicates to provide a more flexible search experience. "
+            "This can be disabled if you need to use a custom Filter.",
+        ),
+    }
+    C4274: dict[str, MessageDefinitionTuple] = {
+        "C4274": (
+            "Don't use custom `search` function, use SearchFilter instead.",
+            "nb-no-search-function",
+            "Nautobot provides a `SearchFilter` class that uses MappedPredicates to provide a more flexible search experience. "
+            "This should be used in place of the custom `search` function.",
+        ),
+    }
+    E4292: dict[str, MessageDefinitionTuple] = {
+        "E4292": (
+            "Inherits from the deprecated StatusModel instead of declaring status on the model explicitly with StatusField",
+            "nb-status-field-instead-of-status-model",
+            "Reference: https://docs.nautobot.com/projects/core/en/next/user-guide/platform-functionality/status/#status-internals",
+        ),
+    }
+    C4701: dict[str, MessageDefinitionTuple] = {
+        "C4701": (
+            "Model's 'app_label.model_name' should be retrieved with 'model._meta.label_lower'",
+            "nb-used-model-label-construction",
+            "Replace f-string '{model._meta.app_label}.{model._meta.model}' with '{model._meta.label_lower}'.",
+        ),
+    }
+    E4281: dict[str, MessageDefinitionTuple] = {
+        "E4281": (
+            "Sub-class name should be %s.",
+            "nb-sub-class-name",
+            "All classes should have a sub-class name that is <model class name><ancestor class type>.",
+        ),
+    }
+    I4282: dict[str, MessageDefinitionTuple] = {
+        "I4282": (
+            "Model was not found in the class.",
+            "nb-no-model-found",
+            "Model was not found in the class.",
+        ),
```

Comment:
Are these out of order? I would have expected them to appear earlier above.

Reply:
I can't remember if the rule number is unique per category or globally, based on this we should sort the ids in a way that helps lookup.

Reply:
These are in numerical order. 

```
The message-id should be a 4-digit number, prefixed with a message category. There are multiple message categories, these being C, W, E, F, R, standing for Convention, Warning, Error, Fatal and Refactoring. The 4 digits should not conflict with existing checkers and the first 2 digits should consistent across the checker (except shared messages).
```

Reply:
Ah, my bad I see that now.